influxql/ast.go

```diff
@@ -952,6 +952,16 @@ func (s *SelectStatement) RewriteWildcards(ic IteratorCreator) (*SelectStatement
 
 	// If there are no dimension wildcards then merge dimensions to fields.
 	if !hasDimensionWildcard {
+		// Remove the dimensions present in the group by so they don't get added as fields.
+		for _, d := range s.Dimensions {
+			switch expr := d.Expr.(type) {
```

Comment:
There's no other type to switch on, but I don't think either is better. I find type switches read easier for me personally so I like those. Since it was already in the function to do it this way, I didn't see a reason to change it.